### DISCUSSION TITLE
Connection Info from IDEA to API

### DISCUSSION THREAD

**Discussion by PippaMorgan** [idea-statica/ideastatica-public#298] [Help Q&A] [answered]
Hello,
I have designed a connection using IDEAStatiCa connection (from Checkbot) and now want to be able to extract the connection information eg, how many bolts, how far apart are they, the size and thicknesses of plates, etc. I'd also like to be able to extract the member names that are being connected. Currently I'm just looking at doing this with a Clip Angle but once I've figured out the process then I'd like to expand to other types too.

Is there a way to extract information about the design? I'd rather not have to set up parameters in IDEA because I'm not using templates so would have to set them up each time

My overall aim is to be able to input the connection into Tekla Structures but I need to be able to extract all of the information from IDEA first.

Thank you in advance for any guidance you might be able to offer,
Pippa

**Comment by nathanlukeIDEA** [comment, marked as answer]
Hi Pippa, 

You can export the connection geometry and information about the connection to the IDEA Open Model Format. From there you can start to decompose the geometry.

You can use the API to do this (using c#):

```csharp

//Starting hidden service for project creation.
string ideaStatiCaDir = "C:\\Program Files\\IDEA StatiCa\\StatiCa 23.0";

string ideaFilePath = ""; //set the filepath to your project here.

ConnHiddenClientFactory calcFactory = new ConnHiddenClientFactory(ideaStatiCaDir);
ConnectionHiddenCheckClient client = calcFactory.Create();

//open your connection file
client.OpenProject(ideaFilePath);

//get first connection
ConProjectInfo conProjectInfo = client.GetProjectInfo();
string connId = conProjectInfo.Connections.First().Identifier;

string ConnectionDataStringXML = client.GetAllConnectionData(connId);

OpenModelContainer container = IdeaRS.OpenModel.Tools.OpenModelContainerFromXml(ConnectionDataStringXML);

OpenModel openModel = container.OpenModel;

int boltGridCount = openModel.Connections.First().BoltGrids.Count;

//etc....

```

If you are using Rhino and Grasshopper, there are implementations that allow you to quickly decompose this information in Grasshopper with Geometry Preview.

![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/5c9d2da9-e208-44a5-805f-43a96d175c24)

Hope this helps.

Cheers
Nathan

**Reply by PippaMorgan** [reply]
Hi Nathan,

Thank you very much for your help and your quick response, it looks to be what I need. I've come across an error when copying the code:

![image](https://github.com/idea-statica/ideastatica-public/assets/137404948/ee02c36d-bf65-417c-84c6-058fdaa64c24)

The pop up error comes from the line above where connectionDataStringXML just shows error. I've made sure that my file paths are correct and as far as I can tell then the other variables have the correct values. Do you know what I can do to solve this issue?

The grasshopper package looks interesting, is it possible to find it anywhere? I have the KarambaIDEA plugin but that looks different.

Thanks
Pippa

**Reply by nathanlukeIDEA** [reply]
Hi, @PippaMorgan just to confirm for everyone, there was an error that was occurring in version 23.0.1 due to the project being created from Tekla and some information not being present. The code is now working when updating to the latest version. Cheers Nathan

**Comment by nathanlukeIDEA** [comment]
Hi @PippaMorgan, 

Would you mind sharing your file with me at nathan.luke@ideastatica.com so I can take a look?

I can also provide you with access to Grasshopper tools through there.

Cheers
Nathan

**Reply by PippaMorgan** [reply]
Hi Nathan,

I've just sent you an email with my VS file and IDEA file.
Thanks for all your help!
Pippa
